oleaut32.go

```diff
@@ -187,6 +198,182 @@ func VariantBSTRToString(value *VAR_BSTR) string {
+}
+
+func (v *VARIANT) GetULVal() (uint32, error) {
+	if v.Vt != VT_UI4 {
+		return 0, fmt.Errorf("Error: LVal() v.Vt !=  VT_UI4, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	return p.ulVal, nil
+}
+
+func (v *VARIANT) SetULVal(value uint32) {
+	v.Vt = VT_UI4
+	p := (*VAR_UI4)(unsafe.Pointer(v))
+	p.ulVal = value
+}
+
+func (v *VARIANT) BoolVal() VARIANT_BOOL {
+	value, err := v.GetBoolVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBoolVal() (VARIANT_BOOL, error) {
+	if v.Vt != VT_BOOL {
+		return VARIANT_FALSE, fmt.Errorf("Error: BoolVal() v.Vt !=  VT_BOOL, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	return p.boolVal, nil
+}
+
+func (v *VARIANT) SetBoolVal(value VARIANT_BOOL) {
+	v.Vt = VT_BOOL
+	p := (*VAR_BOOL)(unsafe.Pointer(v))
+	p.boolVal = value
+}
+
+func (v *VARIANT) BstrVal() *uint16 {
+	value, err := v.GetBstrVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetBstrVal() (*uint16, error) {
+	if v.Vt != VT_BSTR {
+		return nil, fmt.Errorf("Error: BstrVal() v.Vt !=  VT_BSTR, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	return p.bstrVal, nil
+}
+
+func (v *VARIANT) SetBstrVal(value *uint16) {
+	v.Vt = VT_BSTR
+	p := (*VAR_BSTR)(unsafe.Pointer(v))
+	p.bstrVal = value
+}
+
+func (v *VARIANT) PDispVal() *IDispatch {
+	value, err := v.GetPDispVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPDispVal() (*IDispatch, error) {
+	if v.Vt != VT_DISPATCH {
+		return nil, fmt.Errorf("Error: PDispVal() v.Vt !=  VT_DISPATCH, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PDISP)(unsafe.Pointer(v))
+	return p.pdispVal, nil
+}
+
+func (v *VARIANT) SetPDispVal(value *IDispatch) {
+	v.Vt = VT_DISPATCH
+	p := (*VAR_PDISP)(unsafe.Pointer(v))
+	p.pdispVal = value
+}
+
+func (v *VARIANT) PVarVal() *VARIANT {
+	value, err := v.GetPVarVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPVarVal() (*VARIANT, error) {
+	if v.Vt != VT_BYREF|VT_VARIANT {
+		return nil, fmt.Errorf("Error: PVarVal() v.Vt !=  VT_BYREF|VT_VARIANT, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PVAR)(unsafe.Pointer(v))
+	return p.pvarVal, nil
+}
+
+func (v *VARIANT) SetPVarVal(value *VARIANT) {
+	v.Vt = VT_BYREF | VT_VARIANT
+	p := (*VAR_PVAR)(unsafe.Pointer(v))
+	p.pvarVal = value
+}
+
+func (v *VARIANT) PBoolVal() *VARIANT_BOOL {
+	value, err := v.GetPBoolVal()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return value
+}
+
+func (v *VARIANT) GetPBoolVal() (*VARIANT_BOOL, error) {
+	if v.Vt != VT_BYREF|VT_BOOL {
+		return nil, fmt.Errorf("Error: PBoolVal() v.Vt !=  VT_BYREF|VT_BOOL, ptr=%p, value=%+v", v, v)
+	}
+	p := (*VAR_PBOOL)(unsafe.Pointer(v))
+	return p.pboolVal, nil
+}
+
+func (v *VARIANT) SetPBoolVal(value *VARIANT_BOOL) {
+	v.Vt = VT_BYREF | VT_BOOL
+	p := (*VAR_PBOOL)(unsafe.Pointer(v))
+	p.pboolVal = value
+}
+
+func (v *VARIANT) PPDispVal() **IDispatch {
```

Comment:
This should be called `MustPPDispatch` and use `panic` instead of `log.Fatal` to allow recovery.